Source/Charts/Data/Implementations/Standard/ChartDataEntry.swift

```diff
@@ -108,32 +99,18 @@ open class ChartDataEntry: ChartDataEntryBase
     }
 }
 
-public func ==(lhs: ChartDataEntry, rhs: ChartDataEntry) -> Bool
-{
-    if lhs === rhs
-    {
-        return true
-    }
-    
-    if !lhs.isKind(of: type(of: rhs))
+// MARK: Equatable
+extension ChartDataEntry/*: Equatable*/ {
+    public static func ==(lhs: ChartDataEntry, rhs: ChartDataEntry) -> Bool
     {
-        return false
-    }
-    
-    if lhs.data !== rhs.data && !lhs.data!.isEqual(rhs.data)
-    {
-        return false
-    }
-    
-    if fabs(lhs.x - rhs.x) > Double.ulpOfOne
-    {
-        return false
-    }
-    
-    if fabs(lhs.y - rhs.y) > Double.ulpOfOne
-    {
-        return false
+        if lhs === rhs
+        {
+            return true
+        }
+
+        return lhs.isKind(of: type(of: rhs))
+            && (lhs.data?.isEqual(rhs.data) ?? true)
```

Comment:
I am not sure why data is included here. But @danielgindi is missing :(
Why can't we just keep the old conditions like 
```swift
&& lhs.data !== rhs.data
&& !lhs.data!.isEqual(rhs.data)
```
ideally data will not be nil, if it's nil we will get the exception, but the new code will cover it up by returning false